src/engine.ts

```diff
@@ -885,13 +885,17 @@ export default class Engine {
     pl.loadEvents(Event.parse(boardActions[action].income));
     this.endTurnPhase(player, Command.Action);
   }
+  
   [Command.ChooseIncome](player: PlayerEnum, income: string) {
     const { incomes } = this.availableCommand(player, Command.ChooseIncome).data;
-    const spec = '+' + income;
+    const incomeRewards = income.split(",") ;
+
+    for (const incR of incomeRewards) {
+      assert(_.find(incomes, { spec: "+" + incR }), `${incR} is not in the available income`);
+    }
 
-    assert(_.find(incomes, { spec: spec }), `${income} is not in the available income`);
 
-    this.player(player).receiveIncomeEvent(spec);
+    this.player(player).receiveIncomeEvent(Reward.parse(income));
 
     this.selectIncomePhase(player);
```

Comment:
Is it really needed? Maybe it should be `pl.receiveIncome();` instead. So the player does `income t,2pw,t` instead of `income t. income 2pw. income t`

Reply:
Actually it's working the way you are describing. Am I missing something?

Reply:
Let's say that the player has 4 choices: 2t, 1t, 2pw, 1pw.

From what I see, from the current code the player can do several things:

- income 2pw
- income 2pw,1t,2,t
- income 2pw. income 1t,2t
- income 2pw. income 1t. income 2t

If `pl.receiveIncome();` replaces `this.selectIncomePhase(player);`, only the first two lines would be allowed. The game wouldn't ask the player *again* what to do.